server/src/useCases/compliance/handleCreateDraft.ts

```diff
@@ -0,0 +1,61 @@
+import { Request, Response } from 'express';
+import { ComplianceDbRepository, StatusEnum } from "myTypes";
+
+type UploadedFiles = {
+    logo?: Express.Multer.File[];
+    documentation?: Express.Multer.File[];
+};
+
+export default class CreateDraftRequestHandler {
+    private repository: ComplianceDbRepository;
+    private status: StatusEnum;
+
+    constructor(
+        private req: Request,
+        private res: Response,
+        repository: ComplianceDbRepository,
+        status: StatusEnum
+    ) {
+        this.repository = repository;
+        this.status = status;
+    }
+
+    private createFullUrl(path: string): string {
+        return `${this.req.protocol}://${this.req.get('host')}${path}`;
+    }
+
+    async createOrSubmitForm(): Promise<Response> {
+        try {
+            const requiredFields = ['softwareName', 'website', 'documentation', 'description', 'email'];
+            const missingFields = requiredFields.filter(field => !this.req.body[field]);
+            const files = this.req.files as UploadedFiles;
+
+            if (!files.logo || !files.logo[0]) {
+                missingFields.push('logo');
+            }
+            if (missingFields.length > 0) {
+                return this.res.status(400).send(`Missing required fields: ${missingFields.join(', ')}`);
+            }
```

Comment:
shouldn't this validation be done with mongo schema? I don't think it aligns with DRY principle if we have checks both in schema and service, in the future it may case minor issues with updates if fields change. Having one point of truth would be better approach.

Reply:
Thanks for pointing this out, it will be fixed